hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_exchange.py

```diff
@@ -0,0 +1,533 @@
+import asyncio
+import math
+import time
+from decimal import Decimal
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple
+
+from bidict import bidict
+from bxsolana import Provider
+from bxsolana.provider import WsProvider
+from bxsolana_trader_proto import GetMarketsResponse, api
+from bxsolana_trader_proto.api import (
+    GetAccountBalanceResponse,
+    GetQuotesResponse,
+    GetServerTimeResponse,
+    Market,
+    OrderStatus,
+    Side,
+)
+
+from hummingbot.client.hummingbot_application import HummingbotApplication
+from hummingbot.connector.constants import s_decimal_NaN
+from hummingbot.connector.exchange.bloxroute_openbook import (
+    bloxroute_openbook_constants as CONSTANTS,
+    bloxroute_openbook_web_utils as web_utils,
+)
+from hummingbot.connector.exchange.bloxroute_openbook.bloxroute_openbook_api_order_book_data_source import (
+    BloxrouteOpenbookAPIOrderBookDataSource,
+)
+from hummingbot.connector.exchange.bloxroute_openbook.bloxroute_openbook_auth import BloxrouteOpenbookAuth
+from hummingbot.connector.exchange.bloxroute_openbook.bloxroute_openbook_constants import OPENBOOK_PROJECT
+from hummingbot.connector.exchange.bloxroute_openbook.bloxroute_openbook_order_book import BloxrouteOpenbookOrderBook
+from hummingbot.connector.exchange.bloxroute_openbook.bloxroute_openbook_order_data_manager import (
+    BloxrouteOpenbookOrderDataManager,
+)
+from hummingbot.connector.exchange_py_base import ExchangePyBase
+from hummingbot.connector.trading_rule import TradingRule
+from hummingbot.core.data_type.common import OrderType, TradeType
+from hummingbot.core.data_type.in_flight_order import InFlightOrder, OrderState, OrderUpdate, TradeUpdate
+from hummingbot.core.data_type.order_book import OrderBook, OrderBookMessage
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.data_type.trade_fee import AddedToCostTradeFee, DeductedFromReturnsTradeFee, TradeFeeBase
+from hummingbot.core.data_type.user_stream_tracker_data_source import UserStreamTrackerDataSource
+from hummingbot.core.network_iterator import NetworkStatus
+from hummingbot.core.web_assistant.web_assistants_factory import WebAssistantsFactory
+
+if TYPE_CHECKING:
+    from hummingbot.client.config.config_helpers import ClientConfigAdapter
+
+s_logger = None
+
+
+class BloxrouteOpenbookExchange(ExchangePyBase):
+    """
+    BloxrouteOpenbookExchange connects with BloxRoute Labs Solana Trader API provides order book pricing, user account tracking and
```

Comment:
`BloxRoute` => `bloXroute`

Reply:
realized that other exchanges don't even put a comment here